src/sql/workbench/contrib/notebook/browser/notebookEditor.component.ts

```diff
@@ -0,0 +1,197 @@
+		this._register(model.layoutChanged(() => this.detectChanges()));
+
+		this.detectChanges();
+	}
+
+	private async setNotebookManager(): Promise<void> {
+		let providerInfo = await this._notebookParams.providerInfo;
+		for (let providerId of providerInfo.providers) {
+			let notebookManager = await this.notebookService.getOrCreateNotebookManager(providerId, this._notebookParams.notebookUri);
+			this.notebookManagers.push(notebookManager);
+		}
+	}
+
+	private setContextKeyServiceWithProviderId(providerId: string) {
+		let provider = new RawContextKey<string>('providerId', providerId);
+		provider.bindTo(this.contextKeyService);
+	}
+
+	private async awaitNonDefaultProvider(): Promise<void> {
+		// Wait on registration for now. Long-term would be good to cache and refresh
+		await this.notebookService.registrationComplete;
+		this.model.standardKernels = this._notebookParams.input.standardKernels;
+		// Refresh the provider if we had been using default
+		let providerInfo = await this._notebookParams.providerInfo;
+
+		if (DEFAULT_NOTEBOOK_PROVIDER === providerInfo.providerId) {
+			let providers = notebookUtils.getProvidersForFileName(this._notebookParams.notebookUri.fsPath, this.notebookService);
+			let tsqlProvider = providers.find(provider => provider === SQL_NOTEBOOK_PROVIDER);
+			providerInfo.providerId = tsqlProvider ? SQL_NOTEBOOK_PROVIDER : providers[0];
+		}
+	}
+
+	/**
+	 * Get all of the menu contributions that use the ID 'notebook/toolbar'.
+	 * Then, find all groups (currently we don't leverage the contributed
+	 * groups functionality for the notebook toolbar), and fill in the 'primary'
+	 * array with items that don't list a group. Finally, add any actions from
+	 * the primary array to the end of the toolbar.
+	 */
+	private fillInActionsForCurrentContext(): void {
+		let primary: IAction[] = [];
+		let secondary: IAction[] = [];
+		let notebookBarMenu = this.menuService.createMenu(MenuId.NotebookToolbar, this.contextKeyService);
+		let groups = notebookBarMenu.getActions({ arg: null, shouldForwardArgs: true });
+		fillInActions(groups, { primary, secondary }, false, (group: string) => group === undefined || group === '');
+		//this.addPrimaryContributedActions(primary);
+	}
+
+	private get modelFactory(): IModelFactory {
+		if (!this._notebookParams.modelFactory) {
+			this._notebookParams.modelFactory = new ModelFactory(this.instantiationService);
+		}
+		return this._notebookParams.modelFactory;
+	}
+
+	public get model(): NotebookModel | null {
+		return this._model;
+	}
+
+	private isDirty(): boolean {
+		return this._notebookParams.input.isDirty();
+	}
+
+	private handleContentChanged(change: NotebookContentChange) {
+		// Note: for now we just need to set dirty state and refresh the UI.
+		this.detectChanges();
+	}
+
+	public onViewModeChanged(): void {
+		this.viewMode = this.model?.viewMode;
+		this.detectChanges();
+	}
+
+	public getViewMode(): ViewMode {
```

Comment:
It's kind of weird to have this and viewMode both be here but returning slightly different things. For example if something set viewMode to "views" (while the views enabled config was set to false) but then called this function they'd return different values.

This only seems necessary for initializing the value so I'd suggest just putting this logic in the createModelAndLoadContents function (or at least making this private and then making it clear that this is to be used for initialization)